documentstore/domain.py

```diff
@@ -684,3 +684,19 @@ def metrics(self, value: dict):
                 "cannot set metrics with value " '"%s": value must be dict' % value
             ) from None
         self.manifest = BundleManifest.set_metadata(self._manifest, "metrics", value)
+
+    @property
+    def subject_categories(self):
+        return BundleManifest.get_metadata(self.manifest, "subject_categories")
+
+    @subject_categories.setter
+    def subject_categories(self, value: Union[list, tuple]):
+        if not isinstance(value, (list, tuple)):
```

Comment:
Checagem de tipos com `isinstance` é um _code-smell_

Reply:
Qual sua sugestão? Uso de ```try```? 

Repare que nesse caso desejo uma tupla ou uma lista!

Reply:
`value = list(value)`, por exemplo

Reply:
@gustavofonseca 

Sim essa foi a solução inicial, mas gostaria de garantir que não seja algo que implemente a interface de iterator, como por exemplo o  ```str```.

Nesse caso se o usuário passar como parâmetro um  tipo ```str```, iremos ter algo com 

value = ['t', 'e', 's', 't', 'e']

Alguma outra sugestão?

Reply:
Acho bastante perigoso esse desacoplamento para tipos como lista e tuplas!

Reply:
Em conversa com o @gustavofonseca iremos considerar que o cliente precisa garantir os tipos, portanto a abordagem deve ser: 

```value = list(value)```

Reply:
@jamilatta, como eu disse é apenas um _code-smell_ e não um bug ou algo que deve ser necessariamente corrigido. 

A ideia da orientação a objetos, principalmente em linguagens dinâmicas, é que os códigos sejam dependentes de interfaces e não de tipos. A interface de uma sequência, como é o caso de listas e tuplas, prevê minimamente o suporte a `objeto[0]`, i.e., a implementação do método `__getitem__`. Exemplo:

```
>>> class Foo:
...   def __init__(self):
...     self.seq = ["zap", "ploc", "zaz"]
...   def __getitem__(self, index):
...     return self.seq[index]
...
>>>
>>> list(Foo())
['zap', 'ploc', 'zaz']
```

Nem precisa implementar a interface de iterator. Basta o método `__getitem__`.

> Nesse caso se o usuário passar como parâmetro um tipo `str`, iremos ter algo com
> 
> value = ['t', 'e', 's', 't', 'e']

Pode parecer contra intuitivo, mas eu acho que o comportamento está certo. Uma vez que será o cliente o responsável por validar os valores possíveis com o uso de um dicionário controlado (eu imagino, pelo menos), do ponto de vista do kernel os casos abaixo são equivalentes:

```
value = ['t', 'e', 's', 't', 'e']
value = ['fusca', 'gm', 'ferrari', 'fiat', 'alpha']
value = ['biologicas', 'engenharias', 'humanas']
```

Mas novamente, o uso de isinstance não está errado e não precisa ser alterado se essa rigidez for algo avaliado e deliberado por você =]

Reply:
@gustavofonseca 

Legal Gus! 

Como já havíamos definido previamente que a responsabilidade do conteúdo é do cliente, acredito que a abordagem sugerida é mais indicada.

Vou realizar as alterações e ajustes dos testes.

Reply:
@gustavofonseca 

Ajuste realizado!